test/integration/targets/yum/tasks/yum.yml

```diff
@@ -11,8 +11,8 @@
 - name: verify uninstallation of sos
   assert:
     that:
-        - "yum_result.rc == 0"
-        - "rpm_result.rc == 1"
+        - "yum_result is success"
+        - "rpm_result is failed"
```

Comment:
You need to test return code here since you're not testing the result of the `shell` module but the `rc` of the `rpm` command itself.

Reply:
For the `shell` module this is the [same thing](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/commands/command.py#L241) (`command` module [is used](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/shell.py#L18)).